modules/core/src/main/scala/evo/derivation/internal/Matching.scala

```diff
@@ -1,17 +1,38 @@
 package evo.derivation.internal
 
 import scala.deriving.Mirror
-import scala.compiletime.{constValueTuple, summonAll}
+import scala.compiletime.*
 import scala.reflect.ClassTag
 import scala.reflect.TypeTest
+import scala.quoted.*
+
+private inline def summonNamesRecursively[T <: Tuple]: List[String] = inline erasedValue[T] match
+    case _: (t *: ts)  =>
+        summonFrom {
+            case mirror: Mirror.SumOf[`t`]     =>
+                summonNamesRecursively[mirror.MirroredElemTypes] ::: summonNamesRecursively[ts]
+            case mirror: Mirror.ProductOf[`t`] =>
+                constValue[mirror.MirroredLabel] :: summonNamesRecursively[ts]
+        }
+    case _: EmptyTuple => Nil
+
+private inline def summonMatchersRecursively[A, T <: Tuple]: List[TypeTest[A, ?]] = inline erasedValue[T] match
+    case _: (t *: ts)  =>
+        summonFrom {
+            case mirror: Mirror.SumOf[`t`]     =>
+                summonMatchersRecursively[A, mirror.MirroredElemTypes] ::: summonMatchersRecursively[A, ts]
+            case mirror: Mirror.ProductOf[`t`] =>
+                summon[TypeTest[A, t]] :: summonMatchersRecursively[A, ts]
+        }
+    case _: EmptyTuple => Nil
```

Comment:
There are 3 functions like this, similar and at the same type a bit different. I tried, but I failed with generalizing it